src/libsystemd/sd-event/event-util.c

```diff
@@ -172,6 +173,30 @@ int event_add_child_pidref(
         return sd_event_add_child(e, s, pid->pid, options, callback, userdata);
 }
 
+int event_source_get_child_pidref(sd_event_source *s, PidRef *ret) {
+        int r;
+
+        assert(s);
+        assert(ret);
+
+        pid_t pid;
+        r = sd_event_source_get_child_pid(s, &pid);
+        if (r < 0)
+                return r;
+
+        int pidfd = sd_event_source_get_child_pidfd(s);
+        if (pidfd < 0 && !ERRNO_IS_NEG_NOT_SUPPORTED(pidfd))
```

Comment:
So, hrm, I just realized that https://github.com/systemd/systemd/pull/35971/commits/6e14c46bac760d01868b0bf48461f6ac44c86be3 ensures pidfd is always around for child event sources. Hence for new helpers let's not ignore EOPNOTSUPP anymore, which shouldn't get reported in the first place. I'll submit a PR later to port all internals to pidfd-only operations.